File: src/main/java/com/elchologamer/userlogin/listener/JoinQuitListener.java
package com.elchologamer.userlogin.listener;

import com.elchologamer.userlogin.ULPlayer;
import com.elchologamer.userlogin.UserLogin;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

public class JoinQuitListener implements Listener {

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        if (!UserLogin.getPlugin().getConfig().getBoolean("vanillaJoinMessages", !UserLogin.getPlugin().getConfig().getBoolean("disableVanillaJoinMessages", true))) {
            event.setJoinMessage(null);
        }
        ULPlayer.get(event.getPlayer()).onJoin(false);
    }

    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        ULPlayer.get(event.getPlayer()).onQuit();
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/listener/JoinQuitListener.kt
package com.elchologamer.userlogin.listener

import com.elchologamer.userlogin.ULPlayer
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

object JoinQuitListener : Listener {

    @EventHandler
    fun onPlayerJoin(e: PlayerJoinEvent) {
        e.joinMessage = null
        ULPlayer[e.player].onJoin()
    }

    @EventHandler
    fun onPlayerQuit(e: PlayerQuitEvent) {
        ULPlayer[e.player].onQuit()
    }
}